### PR TITLE
Add Soot dependency from the "definitive" build server

### PR DESCRIPTION
**Pull request by bechang**
This is a trial in using the Soot "central" build server.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -56,6 +56,13 @@ lazy val root = (project in file(".")).
       scalaCheck % Test
     ),
 
+    // Soot dependency using Paderborn Nexus
+    resolvers ++= sootResolvers,
+    libraryDependencies += soot,
+    
+    // Alternative Soot dependency using direct nightly build jar
+    // libraryDependencies += soot from "https://soot-build.cs.uni-paderborn.de/nightly/soot/soot-trunk.jar",
+
```
Comment (bechang) [inline build.sbt:65]: I have also tried this alternative that directly links to the nightly build jar, and it also works.

**Comment by bechang** [comment]
@ftc So what I found is that you have use https to reference the build server now. I don't know, but perhaps you could use http in the past.

**Comment by bechang** [comment]
On my machine, I ran into an SSL certificate error:
```
Server access Error: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target url=https://soot-build.cs.uni-paderborn.de/nightly/soot/soot-trunk.jar
```
I had to resolve it by grabbing the certificate from Firefox and then importing into manually into the Java keystore. This seems really yucky, but it looks like it worked fine on Travis. We should find out if this is an issue for others before we merge.

**Comment by bechang** [comment]
To resolve the SSL certificate here, I used a combination of these two pages:

- <https://stackoverflow.com/questions/21076179/pkix-path-building-failed-and-unable-to-find-valid-certification-path-to-requ>
- <https://blog.alwold.com/2011/06/30/how-to-trust-a-certificate-in-java-on-mac-os-x/>

Very sketchy, but it was all that I could find.

**Comment by jpw234** [comment]
Works on my machine

**Comment by octalsrc** [comment]
Compiles and tests successfully for me

**Comment by smover** [comment]
It works on mac

**Comment by jpw234** [comment]
I'm alright with at least attempting to keep up with their nightly build for the moment, so I'll approve this PR

**Review by jpw234 — APPROVED** [review]

**Pull request merged**